esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp

```diff
@@ -1780,7 +1780,7 @@ bool CWsESDLConfigEx::onGetESDLDefinition(IEspContext &context, IEspGetESDLDefin
 
 bool CWsESDLConfigEx::onListESDLDefinitions(IEspContext &context, IEspListESDLDefinitionsRequest&req, IEspListESDLDefinitionsResponse &resp)
 {
-    Owned<IRemoteConnection> conn = querySDS().connect(ESDL_DEFS_ROOT_PATH, myProcessSession(), RTM_LOCK_READ, SDS_LOCK_TIMEOUT_DESDL);
+    Owned<IRemoteConnection> conn = querySDS().connect(ESDL_DEFS_ROOT_PATH, myProcessSession(), RTM_LOCK_READ|RTM_CREATE_QUERY, SDS_LOCK_TIMEOUT_DESDL);
```

Comment:
As per previous comment:
```
Perhaps similar to previous comment, re. best not to overly use RTM_CREATE_QUERY where unnecessary. In this case the non-existence of the node is handled in the same way as it would be if a blank node was created via CREATE_QUERY, so it is prob. best not to add RTM_CREATE_QUERY here.
```